extensions/parquet/table/src/main/java/io/deephaven/parquet/table/pagestore/ColumnChunkPageStore.java

```diff
@@ -48,25 +51,60 @@ private CreatorResult(@NotNull final ColumnChunkPageStore<ATTR> pageStore,
         }
     }
 
+    private static boolean canUseOffsetIndexBasedPageStore(@NotNull final ColumnChunkReader columnChunkReader,
+            @NotNull final ColumnDefinition<?> columnDefinition) {
+        if (columnChunkReader.getOffsetIndex() == null) {
+            return false;
+        }
+        final String version = columnChunkReader.getVersion();
+        if (version == null) {
+            // Parquet file not written by deephaven, can use offset index
+            return true;
+        }
+        // For vector and array column types, versions before 0.31.0 had a bug in offset index calculation, fixed as
+        // part of deephaven-core#4844
```

Comment:
I think we should eventually plan to delete this logic `canUseOffsetIndexBasedPageStore` to workaround bad DHC written parquet. Instead, we should eventually have a more appropriate error message ("please use this tool <link to DHC issue> to re-write the parquet file"). Probably worth filing a new issue to eventually remove this workaround.

Reply:
Created this issue: https://github.com/deephaven/deephaven-core/issues/4879
Will also add a comment in the code.